Modules/_ctypes/_ctypes.c

```diff
@@ -4575,24 +4620,44 @@ PyCFuncPtr_call_lock_held(PyObject *op, PyObject *inargs, PyObject *kwds)
         CDataObject *this;
         this = (CDataObject *)PyTuple_GetItem(inargs, 0); /* borrowed ref! */
         if (!this) {
+            Py_XDECREF(restype);
+            Py_XDECREF(converters);
+            Py_XDECREF(checker);
+            Py_XDECREF(argtypes);
+            Py_XDECREF(errcheck);
```

Comment:
Please switch to `goto error` or `goto finally` at this point.

Reply:
I was avoiding that to keep the diff small for backporting, I have added it now.